src/jit/lower.cpp

```diff
@@ -960,7 +960,16 @@ GenTreePtr Lowering::NewPutArg(GenTreeCall* call, GenTreePtr arg, fgArgTabEntryP
 #endif // FEATURE_MULTIREG_ARGS
 #endif // not defined(FEATURE_UNIX_AMD64_STRUCT_PASSING)
         {
-            putArg = comp->gtNewOperNode(GT_PUTARG_REG, type, arg);
+#ifdef _TARGET_ARM_
+            if (comp->opts.compUseSoftFP && type == TYP_DOUBLE)
+            {
+                putArg = new (comp, GT_PUTARG_REG) GenTreeMultiRegOp(type, arg, nullptr);
```

Comment:
I realize now that `TYP_LONG` is passed using `GT_FIELD_LIST`, so disregard the first question. However, the other comment still applies.

---
In reply to: [123120550](https://github.com/dotnet/coreclr/pull/12360#discussion_r123120550) [](ancestors = 123120550)